lib/async_hooks.js

```diff
@@ -339,10 +322,14 @@ function emitInitS(asyncId, type, triggerId, resource) {
 
 function emitBeforeN(asyncId) {
   processing_hook = true;
-  for (var i = 0; i < active_hooks_array.length; i++) {
-    if (typeof active_hooks_array[i][before_symbol] === 'function') {
-      runCallback(active_hooks_array[i][before_symbol], asyncId);
+  try {
+    for (var i = 0; i < active_hooks_array.length; i++) {
```

Comment:
I don’t know, but I wouldn’t be surprised if this variant would continue to be faster (i.e. I’m reluctant to change this without benchmarking – if you do and you get good results, sure).